uk.co.compucorp.civicrm.hrcore/gulp/tasks/requirejs.js

```diff
@@ -0,0 +1,182 @@
+var gulpSequence = require('gulp-sequence');
+var path = require('path');
+
+var test = require('../test');
+var utils = require('../utils');
+
+var originalExtension;
+
+module.exports = [
+  {
+    name: 'requirejs',
+    fn: requireJsTask
+  },
+  {
+    name: 'requirejs:watch',
+    fn: function (cb) {
+      var extPath, watchPatterns;
+
+      if (utils.canCurrentExtensionRun('requirejs')) {
+        extPath = utils.getExtensionPath();
+        watchPatterns = utils.addExtensionCustomWatchPatternsToDefaultList([
+          path.join(extPath, '**', 'src/**/*.js')
+        ], 'requirejs');
+
+        gulp.watch(watchPatterns, ['requirejs']).on('change', function (file) {
+          if (utils.canCurrentExtensionRun('test')) {
+            try {
+              test.for(file.path);
+            } catch (ex) {
+              test.all();
+            }
+          }
+        });
+        cb();
+      } else {
+        console.log(colors.yellow('Not eligible for this task, skipping...'));
+        cb();
+      }
+    }
+  }
+];
+
+/**
+ * It scans the build.js file of all the CiviHR extension, to check if in any of them
+ * the current extension is marked as a dependency.
+ *
+ * If any are found, then the `requirejs` task is executed for the extension
+ * the build.js file belongs to.
+ *
+ * The task is recursive, stopping when no further dependencies are found
+ *
+ * @param {Function} cb
+ */
+function extensionDependenciesTask (cb) {
+  var buildFiles = find.fileSync(/js(\/[^/]+)?\/build\.js$/, path.join(__dirname, '../../../'));
+
+  var sequence = buildFiles.filter(function (buildFile) {
+    var content = fs.readFileSync(buildFile, 'utf8');
+
+    return (new RegExp(utils.getCurrentExtension(), 'g')).test(content);
+  })
+    .map(function (buildFileWithDependency) {
+      var extension = utils.getExtensionNameFromFile(buildFileWithDependency);
+
+      return utils.spawnTaskForExtension('requirejs', requireJsTask, extension);
+    });
+
+  sequence.length ? gulpSequence.apply(null, sequence)(function () {
+    // Restore the original extension (used in the CLI) as the current extension
+    // before marking the task as done
+    utils.setCurrentExtension(originalExtension);
+
+    cb();
+  }) : cb();
+}
+
+/**
+ * Goes through the given build file content and, for each
+ * extension's placeholder (%{extension-name}) found, it creates an object
+ * with said placholder and the extension's local path
+ *
+ * @param {String} buildFileContent
+ * @return {Array}
+ */
+function getDependencyExtensionsData (buildFileContent) {
+  var matches;
+  var placeholderRegExp = /(%{([^}]+)})/g;
+  var requiredExtensions = [];
+
+  while ((matches = placeholderRegExp.exec(buildFileContent)) !== null) {
+    requiredExtensions.push({
+      placeholder: matches[1],
+      path: utils.getExtensionPath(matches[2])
+    });
+  }
+
+  return requiredExtensions;
+}
+
+/**
+ * Takes the content of the build file on the given path, and applies any
```

Comment:
Please move the part of the sentence after the comma to another line:

Takes the content of the build file on the given path
and applies any required transformations on it